src/main/java/cd/go/authentication/ldap/executor/SearchUserExecutor.java

```diff
@@ -74,10 +73,12 @@ public GoPluginApiResponse execute() throws Exception {
                 LOG.info(String.format("[User Search] Looking up for users matching search_term: `%s`" +
                         " using the search_filter: `%s` and auth_config: `%s`", searchTerm, userSearchFilter, authConfig.getId()));
 
-                List<User> users = ldap.search(userSearchFilter, new String[] {searchTerm}, configuration.getUserMapper(new UsernameResolver()), 100);
+                final List<User> users = ldap.search(userSearchFilter, new String[]{searchTerm}, configuration.getUserMapper(new UsernameResolver()), remainingResultCount);
                 allUsers.addAll(users);
-                if (users.size() == 100)
+
+                if (users.size() >= MAX_SEARCH_RESULT) {
```

Comment:
I think this should be `allUsers.size()`. Consider:

AuthConfig1: Matches 6 users.
AuthConfig2: Matches 6 users.
AuthConfig3: Matches 6 users.

MAX_SEARCH_RESULT = 10.

Ideally, it should not go to `AuthConfig3` at all. But it will.